### ISSUE TITLE
Add an alteration supporting lookups

### ISSUE THREAD

**Issue by Seddryck** [Seddryck/NBi#350]
It should be possible to define an alteration, named lookup, to transform the value of a cell into the value defined in a reference table. This is a bit similar to a join in SQL but at the difference of a join clause when more than row are matching one the reference table, the alteration will fail.

The behaviour should support to discard rows having no match in the reference table or hold them by using a default value or the original value.  The default behaviour should be to fail the alteration.

**Comment by Seddryck** [comment]
Supported syntax
```xml
<alteration>
   <lookup-replace>
      <missing behavior="discard"/>
      <join>
         <mapping candidate="#1" reference="#0"/>
      </join>
      <replacement identifier="#2" />
      <result-set file="reference.csv" />
   </lookup-replace>
</alteration>
```

When using the behavior ```default-value```
```xml
<missing behavior="default-value">Not found!</missing>
```

When using the behavior ```original-value```
```xml
<missing behavior="original-value" />
```

When using the behavior ```failure``` (behaviour by default)
```xml
<missing behavior="failure" />
```

**Comment by Seddryck** [comment]
- [x] Serialization/deserialization
- [x] Engine
- [x] Acceptance testing

**Comment by Seddryck** [comment]
Beta available in 1.21.0-beta0326
